### DISCUSSION TITLE
MET & METplus not working with MacOS

### DISCUSSION THREAD

**Discussion by HathewayWill** [dtcenter/METplus#2018] [Installation] [answered]
Good afternoon,

Having some issues with MET and METplus installing on MacOS.  I know it can be done but I'm not certain what the issue is. Any thoughts @georgemccabe @JohnHalleyGotway @jprestop 

@HathewayWill  My Environment:
- MacOS Ventura 13.2 
- Python 3.10.9 (Homebrew)
- GNU 12.2.0 (Homebrew)

When I use the met_Compile_all.sh the libraries install without a problem, it's the MET installation that has a problem.

I have attached the script that I used for installation as well as all the relevant log files that I could find in a zip file.

Any help would be appreciated as it will add more people who can use the MET tools.

**Comment by jprestop** [comment, marked as answer]
Hi @HathewayWill.  Thank you for all of the files you sent along.

I don't have any experience compiling on a Mac, compiling with a Python version over 3.8.6, or compiling with a GNU version higher than 11, but let's see what we can figure out.

I see the following in your config.log file:
```
ld: unknown option: --disable-new-dtags
```

After googling a bit, I'd like to have you comment out [this line](https://github.com/dtcenter/MET/blob/681dec568ed40bd64d7df63230b8c41df12bda18/internal/scripts/installation/compile_MET_all.sh#L831) in the compile_MET_all.sh script and try running the script again:
```
export LDFLAGS="-Wl,--disable-new-dtags"
```

Please let us know how it goes.

Julie

**Reply by HathewayWill** [reply]
@jprestop 

Good news we got through the configure part of MET now it stops at Make.  But it's progress
[wetransfer_compile_met_all-log_2023-01-23_2328.zip](https://github.com/dtcenter/METplus/files/10484417/wetransfer_compile_met_all-log_2023-01-23_2328.zip)
[make.log](https://github.com/dtcenter/METplus/files/10484433/make.log)

Make log shows that the Python.h doesn't exist but It does here:

/usr/local//Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/include/python3.10/Python.h

**Reply by HathewayWill** [reply]
wills-MacBook-Pro:~ will$ python3-config --cflags

-I/usr/local/opt/python@3.10/Frameworks/Python.framework/Versions/3.10/include/python3.10 -I/usr/local/opt/python@3.10/Frameworks/Python.framework/Versions/3.10/include/python3.10 -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk

wills-MacBook-Pro:~ will$ python3-config --ldflags
-L/usr/local/opt/python@3.10/Frameworks/Python.framework/Versions/3.10/lib/python3.10/config-3.10-darwin -ldl -framework CoreFoundation

**Reply by jprestop** [reply]
Hi @HathewayWill. 

It's good to know there is a Python.h file at the location you listed.  I see a difference in path, with regard to what "python3-config --cflags" is telling you and where the Python.h file is located, so I'm wondering if there is a Python.h file in the /usr/local/opt/python@3.10 area.

I also wonder if there is a libpython file in /usr/local/opt/python@3.10/Frameworks/Python.framework/Versions/3.10/lib/python3.10/config-3.10-darwin.  

Can you please run the following commands and send the output?
```
find /usr/local/opt/python@3.10/ -name Python.h -type f
```
and
```
ls /usr/local/opt/python@3.10/Frameworks/Python.framework/Versions/3.10/lib/python3.10/config-3.10-darwin
```
Thanks!

**Reply by HathewayWill** [reply]
@jprestop 

```
wills-MacBook-Pro:MET-11.0.0 will$ find /usr/local/opt/python@3.10/ -name Python.h -type f
/usr/local/opt/python@3.10//Frameworks/Python.framework/Versions/3.10/include/python3.10/Python.h
```

```
wills-MacBook-Pro:MET-11.0.0 will$ ls /usr/local/opt/python@3.10/Frameworks/Python.framework/Versions/3.10/lib/python3.10/config-3.10-darwin | sort
Makefile
Setup
Setup.local
config.c
config.c.in
install-sh
libpython3.10.a
libpython3.10.dylib
makesetup
python-config.py
python.o

```

@jprestop hope this helps

**Reply by jprestop** [reply]
Hi @HathewayWill.  That does help, thank you!

Could you please set the following values before calling the compile_MET_all.sh script?
```
export MET_PYTHON_CC=-I/usr/local/opt/python@3.10//Frameworks/Python.framework/Versions/3.10/include/python3.10
export MET_PYTHON_LD="-L/usr/local/opt/python@3.10/Frameworks/Python.framework/Versions/3.10/lib/python3.10/config-3.10-darwin -lpython3.10 -ldl -framework CoreFoundation"
```

Please try rerunning with those values and let us know how it goes.

**Reply by HathewayWill** [reply]
@jprestop 

```
make > make.log 2>&1
make install > make_install.log 2>&1
make test > make_test.log 2>&1
Finished compiling at Mon Jan 23 17:05:42 PST 2023

real	15m57.934s
user	11m28.245s
sys	3m47.226s

```

**Reply by jprestop** [reply]
Great! In the directory with the make_test.log file, could you please check for errors by running the following?

> grep -i error make_test.log

**Reply by HathewayWill** [reply]
```
wills-MacBook-Pro:MET-11.0.0 will$ grep -i error make_test.log
wills-MacBook-Pro:MET-11.0.0 will$ 
```

I've run into issues with METplus.  Should I continue it in here or open a new thread?

**Reply by jprestop** [reply]
Great news with no errors, @HathewayWill!

> I've run into issues with METplus. Should I continue it in here or open a new thread?

Yes, please mark this as answered and create a new issue.
